tests/models/test_grad_norm.py

```diff
@@ -0,0 +1,102 @@
+import torch
+import pytest
+import numpy as np
+
+from pytorch_lightning import Trainer
+
+from pytorch_lightning.loggers import LightningLoggerBase
+from pytorch_lightning.utilities import rank_zero_only
+
+from tests.base import EvalModelTemplate
+
+
+class OnlyMetricsListLogger(LightningLoggerBase):
```

Comment:
do we really need this? why is the default logger not applicable here?

Reply:
I have inspected the [LightningLoggerBase](https://github.com/PyTorchLightning/pytorch-lightning/blob/0be530a42774a9a7868ab1f7ba3f6392d6fcde9e/pytorch_lightning/loggers/base.py#L132), and it explicitly declares `log_metrics` abstract. the [DummyLogger](https://github.com/PyTorchLightning/pytorch-lightning/blob/0be530a42774a9a7868ab1f7ba3f6392d6fcde9e/pytorch_lightning/loggers/base.py#L304) completely ignores the data passed to `log_metrics` acting essentially like `... > /dev/null`. I couldn't find any other class derived from `LightningLoggerBase` that would keep list of the data records passed to it, except for the [CustomLogger](https://github.com/PyTorchLightning/pytorch-lightning/blob/0be530a42774a9a7868ab1f7ba3f6392d6fcde9e/tests/loggers/test_base.py#L30) which unfortunately only keeps the most recent records (and is somewhat outside of the package). Thus I simply implemented a special logger for the sole purpose of keeping history of the dicts passed to `log_metrics`.